docs/apis/user.yaml

```diff
@@ -610,6 +612,372 @@ paths:
+        "200":
+          description: User schema updated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UserSchema'
+              example:
+                id: "550e8400-e29b-41d4-a716-446655440000"
+                name: "customer"
+                schema:
+                  email:
+                    type: "string"
+                    unique: true
+                  age:
+                    type: "number"
+                  mobile:
+                    type: "string"
+                  preferredLanguage:
+                    type: "string"
+                    enum: ["en", "es", "fr", "de", "it"]
+                  loyaltyTier:
+                    type: "string"
+                    enum: ["bronze", "silver", "gold", "platinum"]
+        "400":
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              examples:
+                invalid-request-format:
+                  summary: Invalid request format
+                  value:
+                    code: "USR-2001"
+                    message: "Invalid request format"
+                    description: "The request body is malformed or contains invalid data"
+                invalid-attribute-definition:
+                  summary: Invalid attribute definition
+                  value:
+                    code: "USR-2002"
+                    message: "Invalid attribute definition"
+                    description: "One or more attribute definitions are invalid"
+        "404":
+          description: User schema not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "USR-2004"
+                message: "User schema not found"
+                description: "The user schema with the specified ID does not exist"
+        "500":
+          description: Internal server error
+
+    delete:
+      tags:
+        - user-schemas
+      summary: Delete a user type schema by ID
+      description: |
+        Deletes a user type schema. This operation will fail if there are 
+        existing users of this type.
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+            format: uuid
+          example: "550e8400-e29b-41d4-a716-446655440000"
+      responses:
+        "204":
+          description: User schema deleted
+        "400":
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "USR-2005"
+                message: "Cannot delete schema"
+                description: "Cannot delete schema because users of this type exist"
+        "404":
```

Comment:
We don't need 404 response for DELETE requests right? General practice is to return 204 even when the resource is not found

Reply:
Fixed with https://github.com/asgardeo/thunder/compare/91a4bf9364d05bb4ad158a5fa255b34b3cb1681c..2359bb281523048120acb1f8c4d2b4ce08b10f5f